client/src/app/+videos/+video-watch/video-watch.component.ts

```diff
@@ -280,9 +280,18 @@ export class VideoWatchComponent implements OnInit, OnDestroy {
   }
 
   showShareModal () {
+    // Check video was playing before opening support modal
+    const isVideoPlaying = this.isPlaying()
+
     this.pausePlayer()
 
-    this.videoShareModal.show(this.currentTime, this.videoWatchPlaylist.currentPlaylistPosition)
+    const modalRef = this.videoShareModal.show(this.currentTime, this.videoWatchPlaylist.currentPlaylistPosition)
+
+    modalRef.result.then(() => {
+      if (isVideoPlaying) {
+        this.resumePlayer()
+      }
+    })
```

Comment:
Since this is duplicated code from `showSupportModal` (except from line 288 ) I'd suggest to create a method called `showModal` which takes a callback as an argument to show the actual modal.